### PR TITLE
#12145 Update the install documentation.

### LINKED ISSUE #12145: Change documentation for optional dependency SOAPpy to SOAPpy-py3
A minor maintenance issue. [SOAPpy](https://pypi.org/project/SOAPpy/) seems to be no longer maintained and does not support newest Python 3 releases. I've been using this fork instead [SOAPpy-py3](https://pypi.org/project/SOAPpy-py3/) which seems to be actively maintained. While I don't use all the features, the transition was painless and I didn't need to change any of my own code.

For reference, this is the documentation I'm talking about: https://docs.twisted.org/en/latest/installation/howto/optional.html

### PR DESCRIPTION
**Pull request by adiroiban**
## Scope and purpose

Fixes #12145

This updated the documentation about installing Twisted.

We had multiple pages in which the install was documented.
All install related info was moved into a single page.

Redirection were created for the existing page... so we will not have broken links.

THe install process is quite simple and standard, so I think that a single page is good enough.

Also, rather that duplicate the info from pyproject.toml, the documentation only instruct you to check the pyproject.toml file for the actual list of deps.

### How to check

New install doc page

https://twisted--12147.org.readthedocs.build/en/12147/installation.html

---------

Redirecting from previous pages

https://twisted--12147.org.readthedocs.build/en/12147/installation/index.html
https://twisted--12147.org.readthedocs.build/en/12147/installation/howto/optional.html

----------

Link to the documenatation:

The install source code docs now has a link to the new doc page https://github.com/twisted/twisted/blob/12145-optional-install-docs/INSTALL.rst

The README source code also has a link to the new page https://github.com/twisted/twisted/blob/12145-optional-install-docs/README.rst

These linked page is not yet ready.
It will be available once this PR is merged in trunk.

### CODE REVIEW

**Comment by adiroiban** [comment]
needs-review

**Comment by glyph** [comment]
Merging trunk to see if the macOS failure will bother us less

File: docs/installation.rst [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    pip install twisted[tls,http2]
+
+The following optional dependencies are supported:
+
+* **dev** - packages that aid in the development and testing of Twisted itself.
+* **tls** - packages that are needed to work with TLS.
+* **http2** - packages needed for http2 support.
+* **conch** - packages for working with conch/SSH.
+* **serial** - package to work with serial data.
+* **all-non-platform** - installs **tls**, **conch**, **soap**, and **serial** options.
```
Comment (glyph) [inline docs/installation.rst]:
```suggestion
* **all-non-platform** - installs **tls**, **conch**, and **serial** options.
```

**Review by glyph — APPROVED** [review]
Looks like a nice consolidation.

**Comment by glyph** [comment]
> Merging trunk to see if the macOS failure will bother us less

And it did, hooray. Looks like the build really is more reliable now. An actual bug fixed!

**Comment by adiroiban** [comment]
Thanks for the review. I have enabled the auto-merge on this.

Great work for the macOS fix. It was a huge pain .

**Pull request merged**